mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/notation/GenericNotationCombinators.kt

```diff
@@ -0,0 +1,30 @@
+package ch.epfl.sdp.mobile.application.chess.notation
+
+import ch.epfl.sdp.mobile.application.chess.engine.Position
+import ch.epfl.sdp.mobile.application.chess.parser.Combinators.filter
+import ch.epfl.sdp.mobile.application.chess.parser.Combinators.flatMap
+import ch.epfl.sdp.mobile.application.chess.parser.Combinators.map
+import ch.epfl.sdp.mobile.application.chess.parser.Combinators.repeatAtLeast
+import ch.epfl.sdp.mobile.application.chess.parser.Parser
+import ch.epfl.sdp.mobile.application.chess.parser.StringCombinators
+
+/** An object which contains some convenience parser combinators for any notation */
+object GenericNotationCombinators {
+  /** A [Parser] which returns the column in a position. */
+  val column = StringCombinators.char().filter { it in 'a'..'h' }.map { it - 'a' }
+
+  /** A [Parser] which returns the row in a position. */
+  val row = StringCombinators.digit().map { 8 - it }
```

Comment:
If we expose this parser publicly, we should probably check that its output is in the axis bounds.

```kotlin
  val row = StringCombinators.digit().map { 8 - it }.filter { it in 1..Board.Size }
```

Reply:
Addressed in [67a6ba8](https://github.com/epfl-SDP/android/pull/283/commits/67a6ba8e69ab70d1449265ec841c3786d91bddf5).